### PR TITLE
Confirm and log non-queued connections being created during initialization

### PR DESCRIPTION
**Pull request by abencz**

### CODE REVIEW

**Comment by abencz** [comment]
Some background info on this potential bug: https://locusrobotics.atlassian.net/wiki/spaces/~130844200/pages/1473871873/Queue+Manager+Connection+Issues+Notes

And some discussion in Slack: https://locusrobotics.slack.com/archives/G015FB6R13J/p1597695781071400

The basic theory is that a bunch of connections are added between the calls to `_PublisherImpl.__init__` and `_PublisherImpl.set_queue_size`. These connections do not get wrapped by `QueuedConnection` and pollute the connection list.

We can confirm this theory by running this PR at the customer site and observing if we still see non-QueuedConnection connections being added to the connections list for this topic.

File: clients/rospy/src/rospy/topics.py [hunk 1]
```diff
@@ -995,6 +995,9 @@ def add_connection(self, c):
         """
         if self.queue_size is not None:
             c = QueuedConnection(c, self.queue_size)
+        else:
+            import warnings
+            warnings.warn("Adding non-queued connection for topic %s to [%s:%s]" % (self.resolved_name, c.remote_endpoint[0], c.remote_endpoint[1]))
```
Comment (abencz) [inline clients/rospy/src/rospy/topics.py:1000]: Can't use `_logger.warn` before ros is fully initialized or messages will be swallowed.

**Comment by ayrton04** [comment]
So this looks fine, but I wonder if we can get a little bit closer to the action `self.connections` append operation:

https://github.com/locusrobotics/ros_comm/blob/0dd1d9206b6cce3e72cc6fbda29342595dde8fec/clients/rospy/src/rospy/topics.py#L443

Can we check the type of the object that we're about to append there? I know it's extremely unlikely that some other data path exists that we're not seeing, but "extremely unlikely" is how we got here. We might also consider printing out the stack trace if it's the wrong type.

I can deploy this at Boots in the meantime, pending permission.

File: clients/rospy/src/rospy/topics.py (hunk 1 above)
Comment (MJohnson459) [reply clients/rospy/src/rospy/topics.py:1000]: Any idea what logger this uses? The `rospy.topics` logger is set to FATAL by default so we might not see it.
``` 
 -                                                                                                                                                                                                         
    name: "rospy.topics"                                                                                                                                                                                    
    level: "FATAL"

Comment (abencz) [reply clients/rospy/src/rospy/topics.py:1000]: I'm not sure. After normal rospy logging didn't work during testing I copied what I saw elsewhere in the initialization code: https://github.com/locusrobotics/ros_comm/blob/a30195569b4b5029be43eadfb4553462b1af9257/clients/rospy/src/rospy/topics.py#L148

You might be right and this was just a case of me not setting the logging level before running the node 🤦

**Comment by ayrton04** [comment]
@abencz, can we close this, even if we retain the branch for now?

**Comment by ayrton04** [comment]
Are we planning on merging this?

**Comment by abencz** [comment]
> Are we planning on merging this?

No. I'll close the PR.

**Pull request closed without merging**